feat(core): add trigger state filter

MTarek165 opened the pull request:
closes #12570
## Description
- add backend implementation for trigger state filter and add support to it at frontend



https://github.com/user-attachments/assets/0e720459-fc8f-4161-a651-e233157ab3a4

MTarek165 committed:
feat(core): add trigger state filter

MTarek165 committed:
feat(core): add trigger state filter

MTarek165 committed:
feat(ui): add translations for trigger state filter

MTarek165 committed:
feat(ui): add translations for trigger state filter

MTarek165 committed:
fix(core): default to return all triggers if no state match to either enabled or disabled

MTarek165 committed:
fix(core): default to return all triggers if no state match to either enabled or disabled

MTarek165 committed:
refactor(ui): replace search text with drop down select with two states enabled and disabled with adding translations to both

MTarek165 committed:
refactor(ui): replace search text with drop down select with two states enabled and disabled with adding translations to both

Piyush-r-bhaskar reviewed (changes requested):
can you please remove the other translation file and only keep `en.json` ?

Piyush-r-bhaskar commented on ui/src/components/filter/composables/useValues.ts:
```diff
@@ -103,6 +103,10 @@ export function useValues(label: string | undefined, t?: ReturnType<typeof useI1
         STATUSES: buildFromArray(["PENDING", "ACCEPTED", "EXPIRED"]),
         AGGREGATIONS: buildFromArray(["SUM", "AVG", "MIN", "MAX"]),
         RELATIVE_DATE,
+        TRIGGER_STATES:[
+        {label: t("filter.triggerState.enabled"), value: "enabled"},
+        {label: t("filter.triggerState.disabled"), value: "disabled"}
+    ]
```
Looks good🚀

Piyush-r-bhaskar commented on ui/src/components/filter/configurations/triggerFilter.ts:
```diff
@@ -45,6 +45,20 @@ export const useTriggerFilter = (): ComputedRef<FilterConfiguration> => computed
+                    const {VALUES} = useValues("triggers");
+                    return VALUES.TRIGGER_STATES;
+                }
+            },
```
Looks good🚀

MTarek165 committed:
refactor(ui): remove all translations except en.json

MTarek165 committed:
refactor(ui): remove all translations except en.json

MTarek165 commented:
> can you please remove the other translation file and only keep `en.json` ?

done.

MTarek165 commented:
@Piyush-r-bhaskar Is it ok here ?

Piyush-r-bhaskar commented:
Good on FE side but I am waiting my colleagues from BE to give a look in BE code and make EE equivalent change if any

MilosPaunovic commented:
Hey @MTarek165, can you handle the merge conflict here please?

MTarek165 commented:
@MilosPaunovic resolved 👍

MTarek165 committed:
fix(ui): resolve merge conflict at triggerFilter.ts

MilosPaunovic commented:
Hey @loicmathieu, can you check this one again when you get the chance?

loicmathieu commented on jdbc/src/main/java/io/kestra/jdbc/repository/AbstractJdbcRepository.java:
```diff
@@ -455,6 +458,22 @@ private Condition applyKindCondition(Object value, QueryFilter.Op operation) {
+        if (isDisabled == null) {
+            return DSL.noCondition();
+        }
+        return switch (operation) {
+            case EQUALS -> field("value",JSONB.class).contains(JSONB.valueOf("{\"disabled\": " + isDisabled + "}"));
+            case NOT_EQUALS -> field("value",JSONB.class).contains(JSONB.valueOf("{\"disabled\": " + !isDisabled + "}"));
+            default -> throw new InvalidQueryFiltersException("Unsupported operation for Trigger State: " + operation);
+        };
```
This would effectively parser the JSON of the trigger which we tend to avoid.
If you need to select on the `disbaled` field, you need to add a STORED column for it, this is how we always do

MTarek165 replied:
Ok I will add column for it, generally Is the best practice to not read from field from Json?

loicmathieu replied:
Yes, it's always better to read a stored field

loicmathieu commented:
@brian-mulier-p can you please have a quick look at the filter backend impl?

MTarek165 committed:
feat(core): add disabled column for triggers table

MTarek165 committed:
refactor(core): change trigger state implementation to check against disabled column directly instead of json value

MTarek165 commented:
@loicmathieu I have added disabled column and changed the implementation as requested, It works well at postgres and should be same for mysql and h2.
<img width="1300" height="72" alt="migration" src="https://github.com/user-attachments/assets/c1b296bc-4ee9-4a96-b9eb-14d17020d7a0" />
<img width="1400" height="200" alt="database" src="https://github.com/user-attachments/assets/47c52032-6e1f-4514-bfd9-4075a167e568" />

brian-mulier-p commented:
I'll try to have a look tomorrow

MTarek165 commented:
Hi @brian-mulier-p I have applied latest asked changes and it is ok, could you please give final overview to finish this PR?

MTarek165 commented:
@brian-mulier-p please give a look for this PR since it was open from a while.

MilosPaunovic commented:
Maybe @nkwiatkowski could have a look at this, as he's been working on filters recently.

Also, @MTarek165, could you resolve the conflict please?

MTarek165 committed:
chore(system): update triggers_disabled.sql migration versions

brian-mulier-p commented:
Sorry it went out of my mind and idk I didn't get a notification. I'll have a look rn.

brian-mulier-p reviewed (approved):
LGTM I need to do the EE counterpart

MTarek165 commented:
@brian-mulier-p nice, take your time and also may be migration file versions need to be updated to the latest

brian-mulier-p committed:
fix(translations): update translations

brian-mulier-p committed:
fix(core): remove duplicates

brian-mulier-p reviewed (changes requested):
I did the EE counterpart however there seems to be some failing tests. Concurrency-related ones should be ok however there's a searchExecution test failing. Please solve it.

As I'm OOO for the following weeks, any @kestra-io/backend to take over? The EE PR is [here](https://github.com/kestra-io/kestra-ee/pull/6180) so now it's just a matter of having tests passing.

MTarek165 committed:
Merge branch 'develop' into feature/add-trigger-state-filter

MTarek165 commented:
@brian-mulier-p No problem with tests, the tests is unstable only. Look now after merge and running again that the failed tests pass and now only another one test fails, I checked searchExceution() test locally it is passing and anyway it has no relevance to the PR addition, it is ok don't worry.

MilosPaunovic committed:
Merge branch 'develop' into feature/add-trigger-state-filter

MTarek165 commented:
@loicmathieu can you approve and merge this PR since changes and tests are okay on oss version?
 this is to avoid more merge conflicts only, thanks.

loicmathieu reviewed (approved):
LGTM
